internal/handlers/todo_test.go

```diff
@@ -0,0 +1,87 @@
+package handlers_test
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/shreyas1307/go-server/internal/handlers"
+	"github.com/shreyas1307/go-server/internal/store/models"
+	"github.com/stretchr/testify/assert"
+)
+
+type MockServerStore struct{}
+
+func (m *MockServerStore) AddTodo(ctx context.Context, todo models.Todo) error {
```

Comment:
Nice! You're seeing how go handles mocking. I'll teach you something soon about mocks. Once your interface is defined properly, we'll use `go mock` with `go generate` and you won't need to write these yourself 😆